src/main/java/com/karankumar/bookproject/backend/entity/Book.java

```diff
@@ -42,7 +43,9 @@
     @NotNull
     @NotEmpty
     private String title;
+    @Max(value = 23000)
     private Integer numberOfPages;
+    @Max(value = 23000)
```

Comment:
Instead of using the same value twice, could you create a constant (called something like 'max')? Then, we can use this in the test as well if you set the scope to the default package-level access and mark it as `@VisibleForTesting`.

Also, when you create this max variable, could you use an underscore to separate the thousandths (see the [numeric literals section in our styleguide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md#underscores-in-numeric-literals))

Reply:
Surprisingly I can have only constant expression in the annotation, I tried creating a Constant class like

`public final class Constants {
    public static final Integer MAX_PAGES = 23_000;
}
`
and used this in max annotation like 
` @Max(value = Constants.MAX_PAGES)
    private Integer numberOfPages;`

But I am getting java error - java element value must be a constant expression

Reply:
@knjk04 wdyt?

Reply:
If I understood Karan's original comment, he might mean just create a constant like below:

`public static final Integer MAX_PAGES = 23_000;`

in the same class and annotate it with `@VisibleForTesting`

Reply:
`public class Book extends BaseEntity {
    @VisibleForTesting
    public static final Integer MAX_PAGES = 23_000;
    @Max(value = MAX_PAGES)
    private Integer numberOfPages;
    @Max(value = MAX_PAGES)
    private Integer pagesRead;`
In this case as well I get same java error :(

Reply:
Hey how about something like below:

`public static final int MAX_PAGES = 23000
    @NotNull
    @NotEmpty
    private String title;
    @Max(value = MAX_PAGES)
    private Integer numberOfPages;
    @Max(value = MAX_PAGES)`

Reply:
Annotation allows primitive values, changing Integer to int helped.